src/Worker/Core/Shims/TaskOrchestrationContextWrapper.cs

```diff
@@ -14,7 +14,7 @@ namespace Microsoft.DurableTask.Worker.Shims;
 /// </summary>
 sealed partial class TaskOrchestrationContextWrapper : TaskOrchestrationContext
 {
-    readonly Dictionary<string, IEventSource> externalEventSources = new(StringComparer.OrdinalIgnoreCase);
+    readonly Dictionary<string, Queue<IEventSource>> externalEventSources = new(StringComparer.OrdinalIgnoreCase);
```

Comment:
I am not sure what the typical event usage is, but I am going to hazard a guess single events are common. In which case allocating a queue for it is not optimal. We will have to revisit this and see if there is a way to optimize for a few element count cases.

Reply:
Agreed we should revisit this. It turns out we do the same thing with the `NamedQueue<string>` class for buffered events, so we should consider optimizing both.